aqueduct/gateway/views/decorators.py

```diff
@@ -729,3 +732,81 @@ async def wrapper(request: ASGIRequest, *args, **kwargs):
         return await view_func(request, *args, **kwargs)
 
     return wrapper
+
+
+def validate_response_id(view_func):
+    @wraps(view_func)
+    async def wrapper(request: ASGIRequest, response_id: str, *args, **kwargs):
+        token = kwargs.get("token", None)
+        if not token:
+            return JsonResponse({"error": "Invalid request"}, status=400)
+
+        response = get_response_from_cache(response_id)
+        if not response:
+            return JsonResponse({"error": "Response not found"}, status=404)
+
+        if response["email"] != token.user.email:
+            return JsonResponse({"error": "Response not found"}, status=404)
+
+        return await view_func(request, response_id, *args, **kwargs)
+
+    return wrapper
+
+
+def check_tool_availability(view_func):
```

Comment:
Can you add a docstring that summarises what this decorator does, where it can be used, and why?